README.md

```diff
@@ -87,6 +87,50 @@ The PMEM-CSI driver supports running in different modes, which can be controlled
 
 * **_Unified_** mode is intended to run the driver in a single host, mostly for testing the driver in a non-clustered environment.
 
+### Volume Persistency
+
+Normally, volumes are provided by a storage backend that is independent of a particular node. When a node goes offline, the volume can be mounted elsewhere. With PMEM-CSI, volumes are *local* and thus can only be used on the node where they were created.
+
+This limitation needs to be considered when designing and deploying applications that are to use local storage:
+
+* Persistent Volumes
+A volume gets created independently of the application, on some node where there is enough free space. Applications using such a volume are then forced to run on that node and cannot run when the node is
+down. Data is retained until the volume gets deleted.
+* Ephemeral Volumes
+An application is assigned to a node and when it starts to run, a new volume is created for it on that node. When the application stops, the volume is deleted. This [diagram](/docs/images/sequence/pmem-csi-ephemeral-sequence-diagram.png) illustrates how an ephemeral volume gets provisioned in Kubernetes using PMEM-CSI driver.
+* Cache Volumes
+Volumes are created on a certain set of nodes, each with its own local data. Applications are started on those nodes and then get to use the volume on their node. Data persists across application restarts. This
+is useful when the data is only cached information that can be discarded and reconstructed at any time *and* the application can reuse existing local data when restarting. This [diagram](/docs/images/sequence/pmem-csi-cache-sequence-diagram.png) illustrates how a cache volume gets provisioned in Kubernetes using PMEM-CSI driver.
+
+Volume | Kubernetes | PMEM-CSI | Limitations
+--- | --- | --- | ---
+Persistent | supported | supported | topology aware scheduling
+Ephemeral | [in design](https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/20190122-csi-inline-volumes.md#proposal) | supported | topology aware scheduling, resource constraints
+Cache | supported | supported | topology aware scheduling
+
+[Topology aware scheduling](https://github.com/kubernetes/enhancements/issues/490)
+ensures that an application runs on a node where the volume was created. This is obviously needed for persistent volumes, but also for
+delayed persistent volumes because otherwise a restarted application
+might end up on the wrong node. Usage of cache volumes is more flexible,
+but an application still can only run on one of the nodes where the
+a cache volume is available. For CSI-based drivers like PMEM-CSI,
+Kubernetes >= 1.13 is needed. On older Kubernetes releases, pods must
+be scheduled manually onto the right node(s).
```

Comment:
> This is where documentation becomes tricky. Do we also fork the documentation (describe pmem-csi for CSI 1.0 on the master branch and pmem-csi for CSI 0.3 on the older branch) or do we describe both on the same branch?
> 
> Forking documentation would be very confusing for users, we should avoid that.

Motivation behind mainlining specific branches:
-  Currently, I am not clear what parts of this new persistence types we can *really* implement using CSI 0.0.3
- frankly, i felt the code changes required to maintain both versions in one branch are more are less same as mainlining two branches, i.e, mainlining two implementation for every csi service interface and in our splitted controller case 4 branches of controller code.
- Avoid the user to use wrong combinations like csi.0.0.3 + Kuberntes v1.13(+)

Reply:
I don't agree that maintaining two branches will be easier or less confusing for users. But luckily we don't need to discuss this further, because yesterday we agreed that supporting only CSI 1.0 and Kubernetes 1.13 is enough. I had brought up the argument that these different persistency models cannot be implemented well on older Kubernetes due to the lack of topology support and Jim added that we already expect a recent kernel, so also expecting a recent Kubernetes isn't asking too much.